src/models/model.cpp

```diff
@@ -424,6 +433,11 @@ void EnsureDeviceOrtInit(DeviceInterface& device) {
   auto session_options = OrtSessionOptions::Create();
   std::vector<Config::ProviderOptions> provider_options_list;
   provider_options_list.emplace_back(Config::ProviderOptions{device_type_names[static_cast<int>(type)], {}});
+  // QnnHtpShared is a special case. This allocator is only made available when the provider option
+  // 'enable_htp_shared_memory_allocator' is set to 1.
+  if (type == DeviceType::QNN) {
+    provider_options_list.back().options.emplace_back("enable_htp_shared_memory_allocator", "1");
+  }
```

Comment:
We were previously checking for this option based on user settings, now it will always be set?

https://github.com/microsoft/onnxruntime-genai/blob/0d27a3b437c43cc8aa9bae7b8d08fda7fc8ad4e2/src/models/model.cpp#L359

Reply:
Ah, right. I didn't notice it was the code for the trivial session. All good.